pkg/api/types.go

```diff
@@ -530,6 +530,7 @@ type AgentPoolProfile struct {
 	AvailabilityZones                   []string             `json:"availabilityZones,omitempty"`
 	SinglePlacementGroup                *bool                `json:"singlePlacementGroup,omitempty"`
 	VnetCidrs                           []string             `json:"vnetCidrs,omitempty"`
+	PreserveNodesPropertiesInUpgrading  *bool                `json:"preserveNodesPropertiesInUpgrading,omitempty"`
```

Comment:
Does this need to be an optional behavior? I think preserving custom properties should be the default.

Reply:
This option here is mainly to enable AKS to opt-in customers to this new behavior at the beginning. After it is used by some early customers and is good, we can make the new behavior as default. Does it sound good to you?

Reply:
I have a preference for `PreserveNodesProperties`, as it's more readable, and it is more generic. So, for example, if we want to implement a "scale" version of this in the future, we can re-use this.

Reply:
change the new behavior to default in aks-engine, but keep this option for AKS to gradually enable this new behavior.